<h1>Data Science Project: Planning Stage (Individual)</h1>

In [18]:
library(tidyverse)

players_data <- read_csv("https://raw.githubusercontent.com/emmah47/dsci100-project/refs/heads/main/players.csv", show_col_types = FALSE)

<br><br>

<h2>1. Data Description</h2>

In [19]:
head(players_data)

experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

<br>

Below we find some summary statistics and distribution information for variables inside the dataset:

In [24]:
summary(players_data)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 8.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :20.52  
                                       3rd Qu.:22.00  
                                       Max.   :50.00  
                               

In [31]:
print("Player experience summary")
table(players_data$experience)

[1] "Player experience summary"



 Amateur Beginner      Pro  Regular  Veteran 
      63       35       14       36       48 

In [32]:
print("Player subscription summary")
table(players_data$subscribe)

[1] "Player subscription summary"



FALSE  TRUE 
   52   144 

In [33]:
print("Player gender summary")
table(players_data$gender)

[1] "Player gender summary"



          Agender            Female              Male        Non-binary 
                2                37               124                15 
            Other Prefer not to say      Two-Spirited 
                1                11                 6 

In [10]:
# finding rows with missing values
players_data[!complete.cases(players_data), ]

experience subscribe
1 Regular    TRUE     
2 Pro        TRUE     
  hashedEmail                                                      played_hours
1 7d71c49cbbce8dcf0276b2bfecfa2d16f22cb31a4024556d6085884298f2ee8d 0.1         
2 d9473710057f7d42f36570f0be83817a4eea614029ff90cf50d8889cdd729d11 0.2         
  name  gender       Age
1 Devin Two-Spirited NA 
2 Ahmed Other        NA

### Number of Observations:
196

### Number of Variables:
7

### Summary of numerical features: 
**Type**: played_hours and Age has type <dbl>, which is a numeric value that can have decimal points. <br>
**Description**: played hours is the number of hours a player has played the game, age is the age of the player.
| Column       | Min      | 1st Qu.  | Median  | Mean    | 3rd Qu.  | Max     | NA Count |
|-------------|---------|----------|---------|--------|----------|---------|----------|
| played_hours | 0.000  | 0.000    | 0.100   | 5.846  | 0.600    | 223.100 | 0        |
| Age         | 8.00    | 17.00    | 19.00   | 20.52  | 22.00    | 50.00   | 2        |


### Summary of categorical features: 
**Experience Summary** <br>
**Type**: \<chr>, a string<br>
**Description**: the level of previous experience the player has.
| Category  | Count |
|-----------|-------|
| Amateur   | 63    |
| Beginner  | 35    |
| Pro       | 14    |
| Regular   | 36    |
| Veteran   | 48    |

**Subscription Summary**  <br>
**Type**: \<lgl>, true or false boolean value<br>
**Description**: true if the player has subscribed to a game-related newsletter, false if not/
| Subscribed | Count |
|------------|-------|
| FALSE      | 52    |
| TRUE       | 144   |

**Gender Summary**  <br>
**Type**:  \<chr>, a string<br>
**Description**: the player's gender.
| Gender               | Count |
|----------------------|-------|
| Agender             | 2     |
| Female              | 37    |
| Male                | 124   |
| Non-binary          | 15    |
| Other               | 1     |
| Prefer not to say   | 11    |
| Two-Spirited        | 6     |


### Issues with the data:
We have two observations that are missing the "age" feature. We can impute this by using the mean (rounded to an integer) or median of all other ages in the dataset. Another potential issue is that the dataset is very small (196 records), so it may be less representative of the entire population compared to if we had a larget dataset.

### How the data is collected:
When users decide to sign up, they are asked to fill out an anonymized form where they choose a name from a list of availible names and also their experience, gender, age, and email. I'm guessing the played_hours is probably just collected by the researchers logging connections to the server, or some similar method.

<br><br>

<h2>2. Questions</h2>